packages/event-cache/src/backends/PostgreSQLBackend.js

```diff
@@ -0,0 +1,205 @@
+import AbstractBackend from './AbstractBackend'
+
+import { debug } from '../utils'
+
+/**
+ * Convert an event object to an object compatible with sequelize
+ *
+ * @param eventObject {object} to match the event against
+ * @returns {object} object matching the sequelize names
+ */
+function eventToDB(eventObject) {
+  const dbObject = {}
+
+  Object.keys(eventObject).map(key => {
+    // Convert camel to underscore
+    dbObject[
+      key
+        .split(/(?=[A-Z])/)
+        .join('_')
+        .toLowerCase()
+    ] = eventObject[key]
+  })
+
+  // Rework the topic structure to match the DB structure
+  if (dbObject['raw']) {
+    dbObject['data'] = dbObject['raw']['data']
+    if (dbObject['raw']['topics'] instanceof Array) {
+      for (let i = 0; i < dbObject['raw']['topics'].length; i++) {
+        if (i > 3) break
+        dbObject['topic' + i] = dbObject['raw']['topics'][i] || null
+      }
+    }
+    delete dbObject['raw']
+  }
+
```

Comment:
How does the createdAt field get set ?
I think in the old implementation we had it set based on the block timestamp.

Reply:
This was an explicit decision I made, for better or worse.  I did notice that the previous listener set the date, but didn't really think that was important here.  It doesn't match the API of web3's `getPastEvents` or that traditional use case, so I wasn't expecting anyone to query on it.

Right now the backends are also completely oblivious to web3.  To set the actual block stamp, I'll need to fetch the block.  Since the block stamp isn't technically an attribute of an event log, I was hoping not to do this, and that anything that needed to deal with time could fetch the block (or range) itself if necessary before making the request.

So, I guess the question is, is this functionality or data important?  Are there going to be other services/apps that are going to access this table directly?  If so, should I have a unique extra attribute added to these event objects before storage and stripped off upon fetch?

Ideally, or at least I think the intention of this package was to mimic the web3 API and be used in place of the regular un-cached calls.